AmplifyTools/AmplifyXcode/Sources/AmplifyXcode/CLICommandEncodable.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import ArgumentParser
+
+private enum CLICommandEncodableKeys: String, CodingKey {
+    case commandName
+    case abstract
+    case parameters
+}
+
+protocol CLICommandEncodable: Encodable {
+    static var commandName: String { get }
+    static var abstract: String { get }
+    static var paramsRegistry: CLICommandEncodableParametersRegistry { get }
```

Comment:
Do we need an additional wrapping type here, or can we just make `params` a `Set` with direct `get` access?

Reply:
We actually don't need it, I just wanted to add a layer of abstraction and hide the underlying storage, `Set`. (I don't even really like the name :P )